packages/valory/skills/decision_maker_abci/policy.py

```diff
@@ -163,8 +163,22 @@ def has_updated(self) -> bool:
 
     @property
     def random_tool(self) -> str:
-        """Get the name of a tool randomly."""
-        return random.choice(list(self.accuracy_store.keys()))  # nosec
+        """Get the name of a tool randomly between the top 5."""
+
+        # From the accuracy store prepare a list of the top 5 tools
+        # and randomly select one of them
+        tools_with_acc = [
+            (tool, acc_info.accuracy) for tool, acc_info in self.accuracy_store.items()
+        ]
+        tools_with_acc.sort(key=lambda x: x[1], reverse=True)
+        tool_names: List[str] = [tool for tool, _ in tools_with_acc]
+
+        if len(tool_names) > 5:
+            tool_names = tool_names[:5]
+
+        if len(tool_names) == 1:
+            return tool_names[0]
+        return random.choice(tool_names)  # nosec
```

Comment:
I strongly disagree with this approach. This is no more an e greedy policy and the "exploration" is no more exploration but just a random selection among the top 5 performing tools. The whole purpose of the exploration is to search for new, hopefully better, options. We could either:
1. Implement a decay schedule as suggested in the past.
2. Remove the exploration if we do not want it and simply maintain some sort of greedy policy.